formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -21,18 +22,19 @@ import kotlin.reflect.KClass
  * a state change, it will rerun [Formula.evaluate].
  */
 internal class FormulaManagerImpl<Input, State, Output>(
-    val scope: CoroutineScope,
+    internal val formulaTypeKClass: KClass<*>,
+    override val scope: CoroutineScope,
+    override val formulaType: Class<*> = formulaTypeKClass.java,
+    override val onError: (FormulaError) -> Unit,
     val queue: SynchronizedUpdateQueue,
     val batchManager: BatchManager,
     private val delegate: ManagerDelegate,
     private val formula: Formula<Input, State, Output>,
     initialInput: Input,
-    internal val formulaType: KClass<*>,
     private val listeners: Listeners = Listeners(),
     private val inspector: Inspector?,
     val defaultDispatcher: Dispatcher,
-) : FormulaManager<Input, Output>, ManagerDelegate {
-
+) : FormulaManager<Input, Output>, ManagerDelegate, ActionDelegate, EffectDelegate {
```

Comment:
FormulaManagerImpl implements both `ActionDelegate` and `EffectDelegate` and provides the necessary dependencies.